docs/guides/dsm_aws_sobject.md

```diff
@@ -0,0 +1,253 @@
+      access_key = "XXXXXXXXXXXXXXXXXXXX"
+      secret_key = "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"
+      region = "us-east-1"
+      service = "kms"
+    })
+}
+
+// Create a dsm_sobject of type AES key inside DSM
+resource "dsm_sobject" "aes_sobject" {
+  name            = "aes_sobject"
+  obj_type        = "AES"
+  group_id        = dsm_group.normal_group.id
+  key_size        = 256
+  key_ops         = ["EXPORT", "ENCRYPT", "DECRYPT", "WRAPKEY", "UNWRAPKEY", "DERIVEKEY", "MACGENERATE", "MACVERIFY", "APPMANAGEABLE"]
+}
+
+// AWS sobject creation(Copies the key from DSM)
+resource "dsm_aws_sobject" "aws_sobject" {
+  name = "aws_sobject"
+  group_id = dsm_group.aws_group.id
+  description = "AWS sobject"
+  key_ops         = ["EXPORT", "ENCRYPT", "DECRYPT", "WRAPKEY", "UNWRAPKEY", "DERIVEKEY", "MACGENERATE", "MACVERIFY", "APPMANAGEABLE"]
+  key = {
+    kid = dsm_sobject.aes_sobject.id
+  }
+  custom_metadata = {
+    aws-aliases = "dsm_aws_sobject"
+  }
+}
+
+// 1st Rotation of dsm_sobject
+resource "dsm_sobject" "aes_sobject_rotate1" {
+  name            = "aes_sobject"
+  obj_type        = "AES"
+  group_id        = dsm_group.normal_group.id
+  key_size        = 256
+  key_ops         = ["EXPORT", "ENCRYPT", "DECRYPT", "WRAPKEY", "UNWRAPKEY", "DERIVEKEY", "MACGENERATE", "MACVERIFY", "APPMANAGEABLE"]
+  rotate          = "DSM"
+  rotate_from     = dsm_sobject.aes_sobject.name // Name of the old dsm_sobject
+  
+}
+
+// 1st Rotation of dsm_aws_sobject
+resource "dsm_aws_sobject" "aws_sobject_rotate1" {
+  name = "aws_sobject"
+  group_id = dsm_group.aws_group.id
+  description = "AWS sobject"
+  key_ops         = ["EXPORT", "ENCRYPT", "DECRYPT", "WRAPKEY", "UNWRAPKEY", "DERIVEKEY", "MACGENERATE", "MACVERIFY", "APPMANAGEABLE"]
+  key = {
+    kid = dsm_sobject.aes_sobject_rotate1.id // 1st Rotated dsm_sbject
+  }
+  rotate = "DSM"
+  rotate_from = dsm_aws_sobject.aws_sobject.name // Name of the old dsm_aws_sobject
+}
+
+// 2nd Rotation of dsm_sobject
+resource "dsm_sobject" "aes_sobject_rotate2" {
+  name            = "aes_sobject"
+  obj_type        = "AES"
+  group_id        = dsm_group.normal_group.id
+  key_size        = 256
+  key_ops         = ["EXPORT", "ENCRYPT", "DECRYPT", "WRAPKEY", "UNWRAPKEY", "DERIVEKEY", "MACGENERATE", "MACVERIFY", "APPMANAGEABLE"]
+  rotate          = "DSM"
+  rotate_from     = dsm_sobject.aes_sobject1.name // Name of the old dsm_sobject
+
+}
+
+// 2nd Rotation of dsm_aws_sobject
+resource "dsm_aws_sobject" "aws_sobject_rotate2" {
+  name = "aws_sobject"
+  group_id = dsm_group.aws_group.id
+  description = "AWS sobject"
+  key_ops         = ["EXPORT", "ENCRYPT", "DECRYPT", "WRAPKEY", "UNWRAPKEY", "DERIVEKEY", "MACGENERATE", "MACVERIFY", "APPMANAGEABLE"]
+  key = {
+    kid = dsm_sobject.aes_sobject_rotate2.id // 2nd Rotated dsm_sbject
+  }
+  rotate = "DSM"
+  rotate_from = dsm_aws_sobject.aws_sobject_rotate1.name // Name of the old dsm_aws_sobject
+}
+```
+
+## Rotate with AWS Option
+
+```terraform
```

Comment:
1. Indentation has been done via a command(terraform fmt). This command will indent the terraform resources correctly.
2. There are two options here.

- AWS, When the rotate value is AWS, it will invoke the rekey API
- DSM, when the rotate value is DSM, it will invoke the copy API

This is what our UI does today.It applies to azure as well.